simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileLargeDataSets.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.generator;
+
+import static com.hedera.block.simulator.generator.Utils.readFileBytes;
+import static java.lang.System.Logger.Level.INFO;
+
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
+import com.hedera.block.simulator.config.types.GenerationMode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import javax.inject.Inject;
+
+/** A block stream manager that reads blocks from files in a directory. */
+public class BlockAsFileLargeDataSets implements BlockStreamManager {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
```

Comment:
it could be, but why? if is static, I cannot use `getClass().getName(). so we try to make it static only when there is a need to use the logger from another static place, otherwise we can keep it like this, but let me know if there is other advantage to having it static am open to learn.

Reply:
In general, IMO everything should be static first if possible, and final if possible. Static members are a property of the object `Class` and here correct me if I am wrong, but even the system logger is thread safe, so could be shared. Now in this specific case, I agree with you that theoretically this is a singleton, so we do not expect to have more than one logger. But if we have a class that is not a singleton and we instantiate and use a lot, the overhead would be quite big as far as memory and operations needed to instantiate the logger (also something I could be very wrong about, but for some loggers it is problematic to have multiple instances for the same class? I am really not sure about that could be completely wrong about that).  As far as the class name, statically we could access as BlockAsFileLargeDataSets.class.getName() so it is not a problem. This was my general idea and thought process for the static. But for this specific case, consider resolved because of the singleton and what you point out.